### PR TITLE
[TF] Change APIs to use `Int` instead of `Int32`.

### PR DESCRIPTION
**Pull request by dan-zheng**
Friend PR: https://github.com/apple/swift/pull/24012

### CODE REVIEW

File: Sources/DeepLearning/Layer.swift [hunk 1]
```diff
@@ -1275,7 +1273,7 @@ public struct Flatten<Scalar: TensorFlowFloatingPoint>: Layer {
     @differentiable
     public func applied(to input: Tensor<Scalar>, in _: Context) -> Tensor<Scalar> {
         let batchSize = input.shape[0]
-        let remaining = input.shape[1..<input.rank].contiguousSize
+        let remaining = input.shape[1..<Int(input.rank)].contiguousSize
```
Comment (lattner) [inline Sources/DeepLearning/Layer.swift]: Shouldn't .rank return an Int?

**Comment by lattner** [comment]
This looks really great Dan!

File: Sources/DeepLearning/Layer.swift (hunk 1 above)
Comment (dan-zheng) [reply Sources/DeepLearning/Layer.swift]: Done. Pruned all redundant `Int` initializer calls.

**Comment by dan-zheng** [comment]
Ready for review.

File: Sources/DeepLearning/Operators.swift [hunk 2]
```diff
@@ -310,16 +313,15 @@ public extension Tensor where Scalar: FloatingPoint {
...
+        strides: (Int, Int, Int, Int),
         padding: Padding
     ) -> Tensor {
         return Raw.maxPoolV2(
             self,
-            ksize: Tensor<Int32>([kernelSize.0, kernelSize.1,
-                                 kernelSize.2, kernelSize.3]),
-            strides: Tensor<Int32>([strides.0, strides.1,
-                                   strides.2, strides.3]),
+            ksize: Tensor<Int32>([Int32(kernelSize.0), Int32(kernelSize.1), Int32(kernelSize.2), Int32(kernelSize.3)]),
```
Comment (rxwei) [inline Sources/DeepLearning/Operators.swift]: over 100 columns.

Comment (dan-zheng) [reply Sources/DeepLearning/Operators.swift]: Thanks, done.

**Comment by dan-zheng** [comment]
Locally verified that all tests pass. Merging now, then testing/merging https://github.com/apple/swift/pull/24012.

Note: I built a toolchain and `swift build` almost works:
```
$ swift build
Compile Swift Module 'DeepLearning' (8 sources)
/Users/danielzheng/swift-dev/tensorflow-swift-apis/Sources/DeepLearning/Layer.swift:49:35: error: ambiguous use of 'inference'
        return withLearningPhase(.inference) {
                                  ^
/Users/danielzheng/swift-dev/tensorflow-swift-apis/Sources/DeepLearning/Context.swift:28:10: note: found this candidate
    case inference
         ^
TensorFlow.LearningPhase:3:10: note: found this candidate
    case inference
         ^
/Users/danielzheng/swift-dev/tensorflow-swift-apis/Sources/DeepLearning/Layer.swift:60:35: error: ambiguous use of 'inference'
        return withLearningPhase(.inference) {
                                  ^
/Users/danielzheng/swift-dev/tensorflow-swift-apis/Sources/DeepLearning/Context.swift:28:10: note: found this candidate
    case inference
         ^
TensorFlow.LearningPhase:3:10: note: found this candidate
    case inference
         ^
```

If `LearningPhase.inference` is explicitly qualified, `swift build` succeeds.

**Pull request merged**